docs/development-environment.md

```diff
@@ -1,298 +1,444 @@
+* ### Development build
+	The standard development build will create unminified versions of the JavaScript and CSS files. To build Jetpack like this run:
+	
+	```sh
+	yarn build
+	```
+	
+* ### Continuous Development build
+	By default the development build above will run once and if you change any of the files, you need to run `yarn build` again to see the changes on the site. If you want to avoid that, you can run a continous build that will rebuild anytime it sees any changes on your local filesystem. To run it, use:
+	
+	```sh
+	yarn watch
+	```	
+	
+* ### Production build
+	The production build will generate minified files without duplicated code (resulting from dependencies) and will also generate the matching source map and language files. To build it use:
+	
+	```sh
+	NODE_ENV=production yarn build-client
+	```
 
-## Unit-testing the JS code
+### A note on building Jetpack and Node.js versions
 
-Jetpack includes also several [Mocha](https://mochajs.org/) based unit tests.
-To execute them in your local environment, you can use the following commands.
+We try to frequently keep the Node version we use up to date. So, eventually you may need to refresh your package dependencies (i.e., the `node_modules` directories). This is because some dependencies are built specifically for the Node version you used when you installed them (either by running `yarn build` or `yarn`).
 
-### Admin Page unit tests
+We recommend usage of [nvm](https://github.com/nvm-sh/nvm/) for managing different Node versions on the same environment.
 
-Standing on your jetpack directory, run
+**Note:** If you have previously run the Jetpack build tasks (e.g. `yarn build`), and didn't come back to it for a long time, you can
+run this command before building again. Otherwise you may experience errors on the command line while trying to build.
 
 ```
-$ yarn
-$ yarn test-client
-$ yarn test-gui
+$ yarn distclean
 ```
 
-### Jetpack modules unit tests
+---
+<a name="unit-testing"></a>
+# Unit-testing
 
-Standing on your jetpack directory, run
 
-```
-$ yarn
-$ yarn test-modules
-```
+Jetpack includes several [unit tests](https://github.com/Automattic/jetpack/tree/master/tests) that you can run in your local environment before submitting a new Pull Request.
 
-You can also only run tests matching a specific pattern. To do that, use the argument `-g, --grep <pattern>`:
+If you're not familiar with PHP Unit Testing, you can also check [this tutorial](https://pippinsplugins.com/series/unit-tests-wordpress-plugins/)
 
-```
-$ yarn test-gui -g 'my custom pattern to filter tests'
-```
+To get started, there are several ways to run the unit tests, depending on how you set up your development environment.
 
-To use a custom reporter, pass the argument `-R, --reporter <name>`:
+<a name="unit-testing-php"></a>
 
-```
-$ yarn test-client -R 'my_reporter'
-```
+## PHP unit tests
 
-## Installing Composer
+* ### Docker
```

Comment:
The re-organization looks great to me. My only question is whether we should add this here or leave it just in the Docker page...

Reply:
I think having the section here is good. Now that you mention it though, I'm not sure if we should be duplicating instructions between both documents, in case they change. For the moment I think this would be fine and should be updated in future iterations on the documentation.